gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -329,9 +329,9 @@ static std::shared_ptr<Factor> createDiscreteFactor(
 
     // Logspace version of:
     // exp(-factor->error(kEmpty)) / conditional->normalizationConstant();
-    // We take negative of the logNormalizationConstant `log(k)`
-    // to get `log(1/k) = log(\sqrt{|2πΣ|})`.
-    return -factor->error(kEmpty) - conditional->logNormalizationConstant();
+    // negLogConstant gives `-log(k)`
+    // which is `-log(k) = log(1/k) = log(\sqrt{|2πΣ|})`.
+    return -factor->error(kEmpty) + conditional->negLogConstant();
```

Comment:
Red flag. Errors are supposed to be in neg-log-space as well. Same comment about potential bug and missing unit test if not caught.

Reply:
This is fine, since this is the discrete continuous boundary. We want to negate and exponentiate the error to get the discrete factor values. A few lines below, we have the method `probabilitiesFromLogValues` which does the exponentiation.
Alternatively, `probabilitiesFromLogValues` should probably do the negation too.